Cannot use cv2.aruco.Board

### System Information

OpenCV python version: 4.7.0.68
Operating System / Platform: Windows 11
Python version: 3.10.9

### Detailed description

I am porting code from 4.6.0.66 to 4.7.0.68. Aruco has moved from contrib to the main package, and some functions have been renamed. One of these is the constructor for an aruco board. It has been renamed from `cv2.aruco.Board_create` to `cv2.aruco.Board`. Its input arguments have not changed.
However, i am no longer able to call it with the same input arguments as in 4.6.0.66. Even though i put in a list of points and a list of ids of the same size, it always gives me an error that they are not the same length.:
```
cv2.error: OpenCV(4.7.0) D:\a\opencv-python\opencv-python\opencv\modules\objdetect\src\aruco\aruco_board.cpp:160: error: (-215:Assertion failed) ids.size() == objPoints.size() in function 'cv::aruco::Board::Board'
```
Since the signature has not changed, i assume this is some error in the python wrapper or some change internal to the function?

### Steps to reproduce

Code that worked with 4.6.0.66:
```python
# boardCornerPoints is a list of 4x2 float32 matrices, e.g.:
# boardCornerPoints[0] ->
# array([[-178.05125, -151.86725],
#        [-136.15675, -151.86725],
#        [-136.15675, -109.97275],
#        [-178.05125, -109.97275]], dtype=float32)
boardCornerPoints = np.dstack(boardCornerPoints)        # list of 2D arrays -> 3D array
boardCornerPoints = np.rollaxis(boardCornerPoints,-1)   # 4x2xN -> Nx4x2
boardCornerPoints = np.pad(boardCornerPoints,((0,0),(0,0),(0,1)),'constant', constant_values=(0.,0.)) # Nx4x2 -> Nx4x3
cv2.aruco.Board_create(boardCornerPoints, self.aruco_dict, np.array(ids))
```
For 4.7.0.68, the last line should be changed to:
```python
cv2.aruco.Board(boardCornerPoints, self.aruco_dict, np.array(ids))
```
That however does not work, see error above.

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/objdetect/misc/python/test/test_objdetect_aruco.py
+++ b/modules/objdetect/misc/python/test/test_objdetect_aruco.py
@@ -11,6 +11,17 @@ from tests_common import NewOpenCVTests
 
 class aruco_objdetect_test(NewOpenCVTests):
 
+    def test_board(self):
+        p1 = np.array([[0, 0, 0], [0, 1, 0], [1, 1, 0], [1, 0, 0]], dtype=np.float32)
+        p2 = np.array([[1, 0, 0], [1, 1, 0], [2, 1, 0], [2, 0, 0]], dtype=np.float32)
+        objPoints = np.array([p1, p2])
+        dictionary = cv.aruco.getPredefinedDictionary(cv.aruco.DICT_4X4_50)
+        ids = np.array([0, 1])
+
+        board = cv.aruco.Board(objPoints, dictionary, ids)
+        np.testing.assert_array_equal(board.getIds().squeeze(), ids)
+        np.testing.assert_array_equal(np.ravel(np.array(board.getObjPoints())), np.ravel(np.concatenate([p1, p2])))
+
     def test_idsAccessibility(self):
 
         ids = np.arange(17)

--- a/modules/objdetect/src/aruco/aruco_board.cpp
+++ b/modules/objdetect/src/aruco/aruco_board.cpp
@@ -157,7 +157,6 @@ Board::Board():
 
 Board::Board(InputArrayOfArrays objPoints, const Dictionary &dictionary, InputArray ids):
     Board(new Board::Impl(dictionary)) {
-    CV_Assert(ids.size() == objPoints.size());
     CV_Assert(objPoints.total() == ids.total());
     CV_Assert(objPoints.type() == CV_32FC3 || objPoints.type() == CV_32FC1);
 

